src/components/Resource/Forms/Training/TrainingBasicInfo.tsx

```diff
@@ -107,51 +110,107 @@ export function TrainingFormBasic({ model }: { model?: ModelById }) {
+      //
+      //   return {};
+      // });
 
       const versionMutateData = {
         ...(versionId && { id: versionId }),
         modelId: modelId,
         name: modelName,
-        baseModel: 'SD 1.5' as const, // this is not really correct, but it needs something there
+        baseModel: thisModelVersion
+          ? (thisModelVersion.baseModel as BaseModel)
+          : ('SD 1.5' as const), // TODO [bw] this is not really correct, but it needs something there
         trainingStatus: TrainingStatus.Pending,
         trainingDetails: { type: trainingModelType as tmTypes },
       };
 
       upsertVersionMutation.mutate(versionMutateData, {
-        onSuccess: async (vData) => {
-          // TODO [bw] ideally, we would simply update the proper values rather than invalidate to skip the loading step
-          await queryUtils.modelVersion.getById.invalidate({ id: vData.id });
-          await queryUtils.model.getById.invalidate({ id: data.id });
+        onSuccess: async (vResponse) => {
+          queryUtils.training.getModelBasic.setData({ id: modelId }, (old) => {
+            if (!old) return old;
+
+            const versionToUpdate = versionId
+              ? old.modelVersions.find((mv) => mv.id === versionId)
+              : undefined;
+            if (versionToUpdate) {
+              return {
+                ...old,
+                modelVersions: [
+                  {
+                    ...versionToUpdate,
+                    name: modelName,
+                    trainingDetails: {
+                      ...((versionToUpdate.trainingDetails as
+                        | TrainingDetailsObj
+                        | undefined
+                        | null) || {}),
+                      type: trainingModelType,
+                    },
+                  },
+                  ...old.modelVersions.filter((mv) => mv.id !== versionId),
+                ],
+              };
+            } else {
+              return {
+                ...old,
+                modelVersions: [
+                  {
+                    id: vResponse.id,
+                    name: vResponse.name,
+                    baseModel: vResponse.baseModel,
+                    trainingDetails: vResponse.trainingDetails,
+                    trainingStatus: vResponse.trainingStatus,
+                    files: [],
+                  },
+                  ...old.modelVersions,
+                ],
+              };
+            }
+          });
+
+          // TODO [bw] don't invalidate, just update
           await queryUtils.model.getMyTrainingModels.invalidate();
           setAwaitInvalidate(false);
           goNext(modelId, thisStep);
         },
         onError: (error) => {
+          // should we "roll back" the model creation here?
           setAwaitInvalidate(false);
           showErrorNotification({
             error: new Error(error.message),
             title: 'Failed to save model version',
+            autoClose: false,
```

Comment:
I would allow the default `autoClose` value here, any reason why it should be manually closed?

Reply:
i personally find that error messages are good to not autoclose - if you look away for a second you will have no idea why things aren't working. success messages on the other hand are just kind of eye candy and can be autoclosed.